File: go-version/leetcode/LC80/remove_duplicates.go
package lc80

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	curSize, maxSize := 2, len(nums)
	for i := 2; i < maxSize; i++ {
		if nums[i] != nums[curSize-2] {
			nums[curSize] = nums[i]
			curSize++
		}
	}
	return min(curSize, maxSize)
}

File: python-version/LC80/remove_duplicates.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        cur_size = 2
        for i in range(2, len(nums)):
            if nums[i] != nums[cur_size-2]:
                nums[cur_size] = nums[i]
                cur_size += 1
        return min(cur_size, len(nums))